File: Server/src/main/java/jk_5/nailed/server/command/CommandTeam.java
package jk_5.nailed.server.command;

import java.util.*;

import com.google.common.collect.*;

import net.minecraft.command.*;

import jk_5.nailed.api.*;
import jk_5.nailed.api.map.Map;
import jk_5.nailed.api.map.team.*;
import jk_5.nailed.api.player.*;
import jk_5.nailed.util.*;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandTeam extends NailedCommand {

    public CommandTeam() {
        super("team");
    }

    @Override
    public void processCommandWithMap(ICommandSender sender, Map map, String[] args) {
        if(args.length == 0){
            throw new WrongUsageException("/team join <username> <team>");
        }
        if("join".equalsIgnoreCase(args[0])){
            if(args.length == 1){
                throw new WrongUsageException("/team join <username> <team>");
            }
            Player player = NailedAPI.getPlayerRegistry().getPlayer(getTargetPlayer(sender, args[1]));
            if(player == null){
                throw new CommandException("Unknown username " + args[1]);
            }

            if(args.length == 2){
                throw new WrongUsageException("/team join " + args[1] + " <team>");
            }
            Team team = map.getTeamManager().getTeam(args[2]);
            if(team == null){
                throw new CommandException("Unknown team name " + args[2]);
            }

            if(args.length == 3){
                map.getTeamManager().setPlayerTeam(player, team);
                map.broadcastChatMessage(ChatColor.GREEN + "Player " + player.getUsername() + " is now in team " + team.getColoredName());
            }
        }
    }

    @Override
    public List<String> addAutocomplete(ICommandSender sender, String[] args) {
        Map map = NailedAPI.getMapLoader().getMap(sender.getEntityWorld());
        if(args.length == 1){
            return getOptions(args, "join");
        }else if(args.length == 2){
            if("join".equalsIgnoreCase(args[0])){
                return getUsernameOptions(args);
            }
        }else if(args.length == 3){
            if("join".equalsIgnoreCase(args[0])){
                List<String> teams = Lists.newArrayList();
                for(Team team : map.getTeamManager().getTeams()){
                    teams.add(team.getTeamId());
                }
                return getOptions(args, teams);
            }
        }
        return null;
    }
}

File: Server/src/main/scala/jk_5/nailed/server/command/CommandTeam.scala
package jk_5.nailed.server.command

import net.minecraft.command.ICommandSender
import jk_5.nailed.api.map.Map
import jk_5.nailed.util.ScalaUtils._
import jk_5.nailed.api.NailedAPI
import net.minecraft.util.{EnumChatFormatting, ChatComponentText}
import scala.collection.JavaConversions._

/**
 * No description given
 *
 * @author jk-5
 */
object CommandTeam extends ScalaCommand {

  val name = "team"
  val usage = "/team join <username> <team>"

  override def processCommandWithMap(sender: ICommandSender, map: Map, args: Array[String]) = if(args.length != 0) caseInsensitiveMatch(args(0)) {
    case "join" =>
      if(args.length == 1) throw new WrongUsageException("/team join <username> <team>")
      val player = NailedAPI.getPlayerRegistry.getPlayer(NailedCommand.getTargetPlayer(sender, args(1)))
      if(player == null) throw new CommandException("Unknown username " + args(1))
      if(args.length == 2) throw new WrongUsageException(s"/team join ${args(1)} <team>")
      Option(map.getTeamManager.getTeam(args(2))) match {
        case Some(team) =>
          if(args.length == 3){
            map.getTeamManager.setPlayerTeam(player, team)
            val msg = new ChatComponentText(s"Player ${player.getUsername} is now in team ${team.getColoredName}")
            msg.getChatStyle.setColor(EnumChatFormatting.GREEN)
            map.broadcastChatMessage(msg)
          }else throw new WrongUsageException("/team join <username> <team>")
        case None => throw new CommandException("Unknown team name " + args(2))
      }
    case _ => throw new WrongUsageException("/team join <username> <team>")
  }

  override def addAutocomplete(sender: ICommandSender, args: Array[String]): List[String] =
    if(args.length == 1) getOptions(args, "join")
    else if(args.length == 2){
      if(args(0).equalsIgnoreCase("join")) getUsernameOptions(args) else null
    }else if(args.length == 3){
      if(args(0).equalsIgnoreCase("join")){
        getOptions(args, NailedAPI.getMapLoader.getMap(sender.getEntityWorld).getTeamManager.getTeams.map(_.getTeamId))
      }else null
    }else null
}
